analysis/dataflow/state.go

```diff
@@ -82,90 +56,42 @@ type AnalyzerState struct {
 	// a map
 	BoundingInfo BoundingMap
 
-	// Report contains the accumulated report information
-	Report *config.ReportInfo
-
 	reachableFunctions map[*ssa.Function]bool
-
-	// a callgraph computed using the cha analysis. Useful to boostrap the reachable functions
-	chaCallgraph      *callgraph.Graph
-	isReachabilityCha bool
-
-	// Stored errors
-	numAlarms  atomic.Int32
-	errors     map[string][]error
-	errorMutex sync.Mutex
 }
 
-// NewInitializedAnalyzerState runs NewAnalyzerState, and any additional steps that are commonly used in analyses.
+// NewState generates a State from a PointerState
 // This consists in:
-//   - building the ssa program
-//   - running pointer analysis
 //   - computing a map from interface types to the implementations of their methods
 //   - scanning the usage of globals in the program
 //   - linking aliases of bound variables to the closure that binds them
-func NewInitializedAnalyzerState(program *ssa.Program, pkgs []*packages.Package,
-	logger *config.LogGroup, config *config.Config) (*AnalyzerState, error) {
-	program.Build()
-	state, err := NewAnalyzerState(program, pkgs, logger, config, []func(*AnalyzerState){
-		func(s *AnalyzerState) { s.PopulateImplementations() },
-		func(s *AnalyzerState) { s.PopulatePointersVerbose(summaries.IsUserDefinedFunction) },
-		func(s *AnalyzerState) { s.PopulateGlobalsVerbose() },
+//
+// The State returned *does not* have dataflow information computed yet.
```

Comment:
I think one use case would be if we want to run several dataflow analyses (taint, backtrace..) on the same state, and knowing whether the `IntraProceduralPass` has to run or not. Otherwise I don't think there is a use case yet?